src/components/views/admin/BuilderView.tsx

```diff
@@ -1,156 +1,243 @@
+  const handleRatingChange = (
+    memberId: string,
+    newRating: number | null,
+  ): void => {
+    const rateData = appDataArray.find(
+      (data) =>
+        data.data.memberId === memberId && data.type === 'candidateRateInfo',
+    );
+    if (rateData) {
+      handlePatch(rateData.id, {
+        ...rateData.data,
+        rating: newRating,
+        memberId,
+      });
+    } else {
+      handlePost({ memberId, rating: newRating });
+    }
+    setRatings((prevRatings) => ({
+      ...prevRatings,
+      [memberId]: newRating,
+    }));
+  };
+
+  const handleDownload = async (memberId: string): Promise<void> => {
+    const candidateCvData = candidatesCv[memberId];
+    const { component: CvTemplate } = TEMPLATES.find(
+      (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+    ) || { component: FirstTemplate };
+    const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+    const pdfBlob = await pdf(renderedTemplate).toBlob();
+    if (!candidateCvData.cvStatusInfo.customCv) {
+      saveAs(pdfBlob, 'generated-cv.pdf');
+    }
+  };
+  const [visibility, setVisibility] = useState<string | null>(null);
+  const [renderedFile, setRenderedFile] = useState<string | null>(null);
+  const handleView = async (memberId: string): Promise<void> => {
+    if (visibility === memberId) {
+      setVisibility(null);
+      setRenderedFile(null);
+    } else {
+      const candidateCvData = candidatesCv[memberId];
+      const { component: CvTemplate } = TEMPLATES.find(
+        (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+      ) || { component: FirstTemplate };
+      const fileToRender = <CvTemplate cvValues={candidateCvData} />;
+      const pdfBlob = await pdf(fileToRender).toBlob();
+      const dataUrl = URL.createObjectURL(pdfBlob);
+      setVisibility(memberId);
+      setRenderedFile(dataUrl);
+      window.open(dataUrl, '_blank');
+    }
+  };
+  const handleDownloadAll = async (): Promise<void> => {
+    const zip = new JSZip();
+    const candidateIds = Object.keys(candidatesCv);
+
+    // Use Promise.all with map to wait for all async operations to complete
+    await Promise.all(
+      candidateIds.map(async (memberId) => {
+        const candidateCvData = candidatesCv[memberId];
+        const { component: CvTemplate } = TEMPLATES.find(
+          (t) => t.id === candidateCvData.cvStatusInfo.selectedTemplateId,
+        ) || { component: FirstTemplate };
+        const renderedTemplate = <CvTemplate cvValues={candidateCvData} />;
+        const pdfBlob = await pdf(renderedTemplate).toBlob();
+        if (!candidateCvData.cvStatusInfo.customCv) {
+          zip.file(`${memberId}-generated-cv.pdf`, pdfBlob);
+        }
+      }),
+    );
+
+    // Generate the RAR archive and trigger the download
+    zip.generateAsync({ type: 'blob' }).then((content) => {
+      saveAs(content, 'candidates-cv.rar');
```

Comment:
This is very exotic, I would advise to use the `.zip` extension as users might be more familiar with it and the os can extract the archive natively (to the best of my knowledge on macOS and Windows).

Reply:
Yeh of course, `.zip` and other users on other os would be familiar with it, will fix that also.